var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -342,6 +343,16 @@ class IntelOneapiCompilers(IntelOneApiPackage, CompilerPackage):
     # instead.
     requires("%gcc", msg="intel-oneapi-compilers must be installed with %gcc")
 
+    # Hack for working around things like having old binutils in RHEL9.5
+    # https://access.redhat.com/solutions/7049696
+    variant(
+        "binutils",
+        default=False,
+        description="Build a binutils. (Use if unsupported opcode by OS ld/as)",
+    )
+    with when("+binutils"):
+        depends_on("binutils", type=dt.ALL_TYPES)
```

Comment:
So far, our policy has been the following, see [here](https://spack.readthedocs.io/en/latest/getting_started.html#binutils):

> Unlike CMake or Autotools, adding depends_on('binutils') to every package is not considered a best practice because every package written in C/C++/Fortran would need it.

Reply:
I'll check if this works out of the box in #45189, assuming `intel-oneapi-compilers` depends on `gcc+binutils`

Reply:
No, I've tried, I got first compiled `gcc@13.3.0+binutils ^binutils@43`, this gcc is able to pass any vnni test I can think of.

Then installing `intel-oneapi-compilers%gcc@13.3.0` (that gcc@13.3.0, my os gcc is @11), still everywhere uses /usr/bin/ld, e.g. I can grep the log and see /usr/bin/[ar,ld,as,randlib] , so it is not my binutils@43.